File: cleaning/clean_data.py
##
## Transform results data from JSON to CSV
##
## Omar Trejo
## June, 2017
##

import pandas

print("Reading data...")

data = pandas.read_json(
    "~/Downloads/group-14.json"
    encoding="utf-8"
)

print("Saving data...")

data.to_csv(
    "~/Downloads/group-14.csv",
    encoding="utf-8",
    index=False
)

File: cleaning/clean_data.R
##
## Clean data due to unwanted *whitespace* characters
##
## Omar Trejo
## March, 2017
##

library(stringr)

print("Reading data...")

file_name <- "~/Downloads/group-14.csv"

data <- read.csv(
    file_name,
    stringsAsFactors = FALSE,
    na.strings = c("", " ", "NA")
)

print("Cleaning data...")

##
## Seems only `top.nearest.station` and `top.telephone` require cleaning
##
data$top.nearest_station <- gsub("\n", "", data$top.nearest_station)
data$top.telephone       <- gsub("\n", "", data$top.telephone)

for (column in colnames(data)) {
    data[, column] <- gsub("\\s+", " ", str_trim(data[, column]))
}

print("Writing data...")

write.csv(data, file_name, row.names = FALSE)
